orca/utils/flagutils.py

```diff
@@ -24,4 +28,86 @@ def get_bad_ants(date: datetime.date, sources=['AI-VAR', 'AI-LO']):
     res = df[df['source'].isin(sources) & (df['date'] == str(date))]['corr_num'].sort_values().unique().tolist()
     if not res or len(res) == 0:
         raise ValueError(f"No bad antennas found for {date}.")
-    return res
\ No newline at end of file
+    return res
+
+def unpack_flag_metadata(input_file: str, original_shape: tuple) -> np.ndarray:
+    """
+    Unpacks binary flag metadata from a packed binary file into its original FLAG column shape.
+
+    Parameters:
+    -----------
+    input_file : str
+        Path to the binary file containing the packed flag data.
+    original_shape : tuple
+        The shape of the original FLAG array (e.g., (polarizations, channels, rows)).
+
+    Returns:
+    --------
+    np.ndarray
+        A Boolean array of the same shape as the original FLAG column.
+    """
+    if not os.path.isfile(input_file):
+        raise FileNotFoundError(f"The file '{input_file}' does not exist.")
+
+    print(f"Loading packed flags from '{input_file}'...")
+    bit_packed = np.fromfile(input_file, dtype=np.uint8)
+
+    # Calculate the total number of bits that need to be unpacked
+    total_bits = np.prod(original_shape)
+    total_bytes = (total_bits + 7) // 8  # Calculate how many bytes are needed
+
+    if bit_packed.size < total_bytes:
+        raise ValueError(f"Insufficient data in the file. Expected at least {total_bytes} bytes, got {bit_packed.size}.")
+
+    print("Unpacking flags...")
+    flags = np.unpackbits(bit_packed)[:total_bits].astype(bool)  # Only unpack the required number of bits
+    flags = flags.reshape(original_shape)
+
+    print(f"Unpacked FLAG column shape: {flags.shape}")
+    print(f"Total flagged points: {np.sum(flags)}")
+
+    return flags
+
+def plot_flag_metadata_all_polarizations_subplot(flags: np.ndarray, output_dir: str = None) -> None:
```

Comment:
consider saving flagging fractions to a database (clickhouse)? It's a lot of plots I'm not sure if people are gonna look at them.

Reply:
I’m not saving plots for all the files — just added the function so people can look at the saved flags if needed. For now I’m sticking with this instead of using ClickHouse, but will add those later when performance is optimized. I will actually remove flagging fraction since i added it just for testing